scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/InMemoryConnectionStore.java

```diff
@@ -275,6 +276,9 @@ public synchronized boolean put(final Connection connection) {
 			}
 			ConnectionId connectionId = connection.getConnectionId();
 			if (connectionId == null) {
+				if (connectionIdGenerator == null) {
+					throw new IllegalStateException("Connection id generator must be attached before!");
+				}
```

Comment:
should we move this check in newConnectioId() ?

Reply:
My idea was, to hide the private `newConnectioId` and report the public `put`. For me this makes it easier to document the requirements.

Reply:
I thought that it would be better to get the check closer from where we use `connectionIdGenerator` attribute.
But we can let it like this if you prefer.